### PR TITLE
Type alias for codec and minor improvments

### PR DESCRIPTION
**Pull request by Liamsi**
Introduce a `Codec` type and go fmt (hence review with "Hide whitespace changes" in the PR diff view: https://github.com/LazyLedger/rsmt2d/pull/1/files?diff=unified&w=1)

### CODE REVIEW

File: datasquare.go [hunk 1]
```diff
@@ -1,242 +1,248 @@
...
 }
 
 func (ds *dataSquare) computeRowProof(x uint, y uint) ([]byte, [][]byte, uint, uint) {
-    tree := merkletree.New(ds.hasher)
-    tree.SetIndex(uint64(y))
-    data := ds.Row(x)
+	tree := merkletree.New(ds.hasher)
+	err := tree.SetIndex(uint64(y))
+	if err != nil {
+		panic(err)
```
Comment (musalbas) [inline datasquare.go:196]: Shouldn't we pass the error upstream if an invalid tree index is supplied?

Comment (Liamsi) [reply datasquare.go:196]: Yes, absolutely! Good catch. All other places were in tests where it was OK to panic I guess. Will update in a bit.

File: datasquare.go [hunk 2]
```diff
@@ -1,242 +1,248 @@
...
 
 func (ds *dataSquare) computeColumnProof(x uint, y uint) ([]byte, [][]byte, uint, uint) {
-    tree := merkletree.New(ds.hasher)
-    tree.SetIndex(uint64(x))
-    data := ds.Column(y)
+	tree := merkletree.New(ds.hasher)
+	err := tree.SetIndex(uint64(x))
+	if err != nil {
+		panic(err)
+	}
```
Comment (adlerjohn) [inline datasquare.go:213]: Here too?

Comment (Liamsi) [reply datasquare.go]: done in 0a13699

**Comment by Liamsi** [comment]
Added some more error handling gotchas in dee82e7

**Pull request merged**